modules/11_subclustering/subclustering.qmd

```diff
@@ -0,0 +1,676 @@
+# - Barcodes that are not part of the default assay will be set to NA
+sc$merged_clusters = ifelse(SeuratObject::Cells(sc) %in% bcs, 
+                            sc[[cluster_column, drop=TRUE]] %>% as.character(), 
+                            NA)
+
+# Specify what to merge/rename
+n = sc[[]] %>% 
+  dplyr::filter(merged_clusters %in% c("1", "2", "3")) %>%
+  rownames()
+sc$merged_clusters[n] = "1-3"
+
+n = sc[[]] %>% 
+  dplyr::filter(merged_clusters %in% c("4", "5", "6")) %>%
+  rownames()
+sc$merged_clusters[n] = "4-6"
+
+# Make a table what was merged/renamed
+merge_rename_results = sc[[c(cluster_column, "merged_clusters")]]
+merge_rename_results = merge_rename_results[bcs, ]
+merge_rename_results = merge_rename_results %>% 
+  dplyr::count(!!sym(cluster_column), merged_clusters)
+
+gt::gt(merge_rename_results)
+```
+
+```{r}
+#| label: fig-subclustering_merge_rename_cluster_2
+#| fig-cap: 'Clusters after merging and/or renaming'
+#| eval: false
+
+# Plot a UMAP with the merged/renamed clusters
+DimPlot(sc, group.by="merged_clusters", label=TRUE) + 
+  AddPlotStyle()
+```
+
+-->
+
+## Subclustering
+
+USER_INPUT:
+
+- Remove the HTML comments
+- Set eval in the chunk options to true
+
+```{r}
+#| label: subclustering_split_cluster_1
+#| eval: true
+
+# Chunk initalizes the subclustering column. Needs to be run only once at the beginning.
+
+# USER_INPUT: 
+# - Set the name of the column containing the clusters to be subclustered
+cluster_column = "seurat_clusters"
+
+# Get barcodes for default assay
+bcs = SeuratObject::Cells(sc[[default_assay]])
+
+# Initialize the column for subclustering but only for the barcodes of the affected assay
+sc$subclustering = ifelse(SeuratObject::Cells(sc) %in% bcs, 
+                            sc[[cluster_column, drop=TRUE]] %>% as.character(), 
+                            NA)
+
+# Set idents to column "subclustering"
+SeuratObject::Idents(sc) = "subclustering"
+```
+
+<!--
+### Simple - Cluster 10
+
+USER_INPUT:
+
+- Remove the HTML comments
+- Set eval in the chunk options to true
+- Add explanation which cluster is split into what
+
+```{r}
+#| label: subclustering_split_cluster_2
+#| eval: false
+
+# Simple subclustering: Use the existing neighbors graph and split a cluster into subclusters. This method is the fastest. However, note that the neighbors graph is based on the dimensionality reduction of the top variable features of the WHOLE dataset. It might not be suitable for more fine-grained substructures within clusters. In such cases, a full subclustering of the cluster should be performed. 
+
+# USER_INPUT:
+# - Set the name of the cluster to be split
+# - Set the name of the neighbors graph that was used for clustering
+#  (see sc@commands$FindClusters, typically "RNA_snn" or "RNA_pca_snn")
+# - Set the subclustering resolution
+# - Provide a list of markers to verify the results 
+cluster_to_split = "10"
+graph_name = "RNA_snn"
```

Comment:
Just to make sure > is this still correct? I remember at some point in the code you removed "_".